src/io.jl

```diff
@@ -57,6 +58,11 @@ function create_internal_structures(connection)
 
     timeframe = Timeframe(num_periods.period, TulipaIO.get_table(connection, "rep_periods_mapping"))
 
+    groups = [
```

Comment:
Can this be done more elegantly with a `collect` or `|>`?

Reply:
I'm not sure. I tried, but it failed (maybe because I'm still understanding what collect does). After I collected it and tried to pass it to the Group...

```julia
TulipaIO.get_table(Val(:raw), connection, "groups_data") |> collect |> TulipaEnergyModel.Group
```

The code I used is basically a replica of the code for the representative periods. I was also trying to maintain consistency in our code. But, I'm open to more elegant ways to code @abelsiqueira. Do you have any ideas or suggestions from your side?

Reply:
You can do this with

```
groups = TIO.get_table(...) .|> row -> Group(row...)
```

but I find it harder to read and understand a posteriori. Instead we can use the `Group(row...)` in the same construct as below. I'll suggest below.